modular_skyrat/modules/deployables/code/deployable.dm

```diff
@@ -0,0 +1,736 @@
+	enabled_function = NONE
+
+/obj/structure/bed/borg_action_pacifier/proc/lock(mob/living/clicker)
+	if(clicker)
+		log_combat(clicker, buckled_cyborg, "locked down cyborg")
+		log_silicon("[key_name(clicker)] locked down [key_name(buckled_cyborg)].")
+
+	playsound(src, 'modular_skyrat/master_files/sound/effects/robot_lock.ogg', 50, TRUE, falloff_exponent = 10)
+	buckled_cyborg.SetLockdown(TRUE)
+	locked = TRUE
+	render_lock()
+
+/obj/structure/bed/borg_action_pacifier/proc/unlock(mob/living/clicker)
+	if(clicker)
+		log_combat(clicker, buckled_cyborg, "released cyborg")
+		log_silicon("[key_name(clicker)] released [key_name(buckled_cyborg)] from lockdown.")
+
+	buckled_cyborg.SetLockdown(FALSE)
+	buckled_cyborg.regenerate_icons()
+	locked = FALSE
+
+/obj/structure/bed/borg_action_pacifier/proc/render_lock()
+	// Emergency lights which are otherwise shamefully unused
+	buckled_cyborg.cut_overlay(buckled_cyborg.eye_lights)
+	buckled_cyborg.eye_lights = new()
+	buckled_cyborg.eye_lights.icon_state = "[buckled_cyborg.model.cyborg_base_icon]_e_r"
+	buckled_cyborg.eye_lights.plane = ABOVE_LIGHTING_PLANE
+	buckled_cyborg.eye_lights.icon = buckled_cyborg.icon
+	buckled_cyborg.add_overlay(buckled_cyborg.eye_lights)
+
+/obj/structure/bed/borg_action_pacifier/proc/undeploy(mob/living/clicker)
+	if(!deployed)
+		return
+	if(locked)
+		unlock()
+	if(buckled_cyborg)
+		unbuckle_mob(buckled_cyborg, TRUE)
+
+	addtimer(CALLBACK(src, .proc/finish_undeploy), SETUP_TIME)
+	playsound(src, 'modular_skyrat/master_files/sound/effects/robot_trap.ogg', 25, TRUE, falloff_exponent = 10)
+	flick("undeploying", src)
+
+/obj/structure/bed/borg_action_pacifier/proc/finish_undeploy()
+	var/obj/item/grenade/borg_action_pacifier_grenade/BAPer = new (get_turf(src))
+	BAPer.power_storage = power_storage
+	BAPer.balloon_alert_to_viewers("reset")
+	if(obj_flags & EMAGGED)
+		BAPer.obj_flags |= EMAGGED
+	if(obj_flags & EMPED)
+		BAPer.obj_flags |= EMPED
+		if(BAPer.power_storage)
+			BAPer.alt_clicked_grenade()
+	qdel(src)
+
+// Buckle overwrites
+/obj/structure/bed/borg_action_pacifier/buckle_mob(mob/living/target, force, check_loc)
+	if(!deployed)
+		return
+	..()
+
+/obj/structure/bed/borg_action_pacifier/user_buckle_mob(mob/living/target, mob/user, check_loc)
+	if(!target || !user)
+		return
+
+	if(target && (target != user))
+		user.visible_message(span_warning("[user] starts buckling [target] to [src]!"))
+	else
+		target.visible_message(span_warning("[target] starts buckling [target.p_them()]self to [src]!"))
+		if(!do_after(target, 1.5 SECONDS, src)) // The added delay is to prevent accidental buckling
+			return
+	..()
+
+/obj/structure/bed/borg_action_pacifier/unbuckle_mob(mob/living/buckled_mob, force, can_fall)
+	if(!force)
+		return
+	..()
+
+/obj/structure/bed/borg_action_pacifier/post_buckle_mob(mob/living/target)
+	if(iscarbon(target))
+		var/mob/living/carbon/carbon = target
+		carbon.set_lying_angle(0)
```

Comment:
This isn't early returning or anything, so it's gonna continue the proc assuming the carbon is a cyborg, and use vars carbons don't have